plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java

```diff
@@ -292,4 +333,70 @@ private static Block getColumnMetricBlock(RowType columnMetricType, Object min,
         rowBlockBuilder.closeEntry();
         return columnMetricType.getObject(rowBlockBuilder, 0);
     }
+
+    private static class StructLikeWrapperWithStructType
```

Comment:
Lets rename this, now that the field is `fieldIdToIndex` instead of the `structType` itself